crates/aiken-lang/src/parser/literal/int.rs

```diff
@@ -6,5 +6,16 @@ use crate::parser::{
 };
 
 pub fn parser() -> impl Parser<Token, (String, Base), Error = ParseError> {
-    select! { Token::Int {value, base} => (value, base)}
+    choice((just(Token::NewLineMinus), just(Token::Minus)))
```

Comment:
Actual fix is here ☝️, had to introduce this `NewLineMinus` fellow to preserve the newline semantic of the when-clauses that's in the end, quite essential to the pattern parsing. Basically everything now will expect a `Minus` or `NewLineMinus` except bin ops, that only operate on `Minus`. This removes the ambiguity when a pattern on the next line starts with a `-` (and would be mistakenly interpreted as a continuation of the expression of the previous when branch). 

I tried different things with `rewind` and al, but would always find an edge case that made it difficult. The newline token does the trick once again.